src/main/java/org/commcare/formplayer/screens/FormplayerSyncScreen.java

```diff
@@ -32,38 +34,27 @@ public void init (SessionWrapper sessionWrapper) throws CommCareSessionException
         Entry commandEntry = sessionWrapper.getPlatform().getEntry(command);
         if (commandEntry instanceof RemoteRequestEntry) {
             PostRequest syncPost = ((RemoteRequestEntry)commandEntry).getPostRequest();
-            setUrl(syncPost.getUrl().toString());
+            url = syncPost.getUrl().toString();
             Hashtable<String, String> params = syncPost.getEvaluatedParams(sessionWrapper.getEvaluationContext());
-            UriComponentsBuilder builder = UriComponentsBuilder.newInstance();
-            // hack because HQ isn't accepting the first query param key properly
-            builder.queryParam("buffer", "buffer");
```

Comment:
Does one of your changes make this no longer necessary?

Reply:
This was never really necessary, it was an artifact of the way this data was being sent (a query string masquerading as form data). Because this is using a query builder the data is rendered as `?buffer=buffer&....`. On CommCare's side the first param then is `?buffer` (note the question mark) which is why this had to be added. Now that we're letting the framework serialize the data it is coming through correctly (without the question mark).